src/java/org/apache/cassandra/db/compaction/ActiveOperations.java

```diff
@@ -49,13 +76,41 @@
     @Override
     public NonThrowingCloseable onOperationStart(TableOperation op)
     {
-            operations.add(op);
-            return () -> {
-                operations.remove(op);
-                TableOperation.Progress progress = op.getProgress();
-                CompactionManager.instance.getMetrics().bytesCompacted.inc(progress.total());
-                CompactionManager.instance.getMetrics().totalCompactionsCompleted.mark();
-            };
+        AbstractTableOperation.OperationProgress progress = op.getProgress();
+        for (CompactionProgressListener listener : listeners)
+        {
+            try
+            {
+                listener.onStarted(progress);
+            }
+            catch (Throwable t)
+            {
+                String listenerName = listener.getClass().getName();
+                logger.error("Unable to notify listener {} while trying to start compaction {} on table {}",
+                             listenerName, progress.operationType(), progress.metadata(), t);
+            }
+        }
+        operations.add(op);
+        return () -> {
+            operations.remove(op);
+            AbstractTableOperation.OperationProgress progressOnCompleted = op.getProgress();
```

Comment:
should we port `totalBytesScanned` into `OperationProgress`?

Reply:
@jasonstack How is it related to this PR? I don't see that `totalBytesScanned` is part of DSE and I don't see where it is used.

Reply:
`RepairMetrics` uses `progressOnCompleted.totalScanned()` and it's not available in OSS. `totalBytesScanned` was added together with `CompactionProgressListener` in the same BDP PR

Reply:
Ported in https://github.com/datastax/cassandra/pull/390/commits/cc7a3660ced952d13b9abe17c80455fb73674640